src/NLog/LayoutRenderers/LayoutRenderer.cs

```diff
@@ -304,5 +310,15 @@ public static void Register(string name, Func<LogEventInfo, LoggingConfiguration
             
             ConfigurationItemFactory.Default.GetLayoutRenderers().RegisterFuncLayout(name, layoutRenderer);
         }
+
+        /// <summary>
+        /// Resolve from DI
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        protected T Resolve<T>() where T : class
```

Comment:
Think I will skip this, until we have figured out a good way to do this.

Reply:
What isn't good about this?

Reply:
I was think to:

support Ctor injection and the Resolve method for Layouts/target etc. The user could choose. For the NLog components we will use the Resolve Method.